src/modules/BuilderPlace/actions/discord.ts

```diff
@@ -0,0 +1,148 @@
+import { getAllowedTokenList } from "../../../queries/global";
+import { IToken } from "../../../types";
+import { ZERO_ADDRESS } from "../../../utils/constant";
+import { parseRateAmount } from "../../../utils/currency";
+import { readFromIPFS } from "../../../utils/ipfs";
+import { truncate } from "../../Messaging/utils/messaging";
+import { getBuilderPlaceById, getDiscordWebhookDetailsByBuilderPlaceId } from "./builderPlace";
+
+export const removeMarkdown = (markdownString: string) => {
+  // Remove headings (##, ###, ####, etc.)
+  markdownString = markdownString.replace(/#{1,6}\s/g, '');
+
+  // Remove emphasis (*) and bold (**)
+  markdownString = markdownString.replace(/[*]{1,2}/g, '');
+
+  // Remove code blocks (```)
+  markdownString = markdownString.replace(/`{3}.*`{3}/g, '');
+
+  // Remove inline code (`)
+  markdownString = markdownString.replace(/`{1}.*`{1}/g, '');
+
+  // Remove links ([...](...))
+  markdownString = markdownString.replace(/\[([^\]]+)\]\([^\)]+\)/g, '$1');
+
+  // Remove images (![...](...))
+  markdownString = markdownString.replace(/!\[([^\]]+)\]\([^\)]+\)/g, '$1');
+
+  return markdownString;
+}
+
+export const sendWelcomeMessage = async (builderPlaceId: string, newWebhookUrl: string) => {
+  const webhookDetails = await getDiscordWebhookDetailsByBuilderPlaceId(builderPlaceId)
+  console.log(webhookDetails, newWebhookUrl, "ah")
+
+  if (webhookDetails?.discordWebhook === newWebhookUrl) return;
+  console.log("sendit", newWebhookUrl, "ah")
+
+  const embed = {
+    title: 'Congratulations!',
+    description: 'BuilderPlace notification is set up. I will notify you here for each new service created!',
+    color: 0x00ff00,
+    author: {
+      name: 'BuilderPlace',
+      icon_url: 'https://builder.place/logo.png',
+    },
+  };
+
+  const payload = {
+    content: ' ', // Empty content to trigger the embed
+    embeds: [embed],
+  };
+
+  fetch(newWebhookUrl, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(payload),
+  })
+    .then((response) => {
+      if (response.ok) {
+        console.log('Message sent successfully!');
+      } else {
+        console.error('Error sending message:', response.statusText);
+      }
+    })
+    .catch((error) => {
+      console.error('Error:', error.message);
+    });
+}
+
+
+export const sendNewServiceNotification = async (builderPlaceId: string, serviceId: string, cid: string) => {
+  const service: { title: string, about: string, keywords: string, rateToken: string, rateAmount: string } = await readFromIPFS(cid);
```

Comment:
in term of architecture, we now make any create service API post success depending on an IPFS call, and discord call. This is not good for maintability. Would be better to imagine something non bloquant with a message broker by example. Any suggestion?